src/libusermode/userhook.cpp

```diff
@@ -1003,12 +1015,15 @@ void drakvuf_load_dll_hook_config(drakvuf_t drakvuf, const char* dll_hooks_list_
         std::vector<std::unique_ptr<ArgumentPrinter>> arg_vec1;
         arg_vec1.push_back(std::make_unique<ArgumentPrinter>("wVersionRequired", config));
         arg_vec1.push_back(std::make_unique<ArgumentPrinter>("lpWSAData", config));
-        wanted_hooks->emplace_back("ws2_32.dll", "WSAStartup", log_and_stack, std::move(arg_vec1));
+        plugin_target_config_entry_t e1("ws2_32.dll", "WSAStartup", log_and_stack, std::move(arg_vec1));
 
         std::vector<std::unique_ptr<ArgumentPrinter>> arg_vec2;
         arg_vec2.push_back(std::make_unique<ArgumentPrinter>("ExitCode", config));
         arg_vec2.push_back(std::make_unique<ArgumentPrinter>("Unknown", config));
-        wanted_hooks->emplace_back("ntdll.dll", "RtlExitUserProcess", log_and_stack, std::move(arg_vec2));
+        plugin_target_config_entry_t e2("ntdll.dll", "RtlExitUserProcess", log_and_stack, std::move(arg_vec2));
+
+        if (!hook_filter(e1)) wanted_hooks[e1.dll_name].emplace_back(std::move(e1));
+        if (!hook_filter(e2)) wanted_hooks[e2.dll_name].emplace_back(std::move(e2));
```

Comment:
I'm not 99% sure you can use `wanted_hook[e1.dll_name]` followed by `std::move(e1)` (as you first fetch the element from the map)`, but would like to verify 1 more time.

Also, this is a semi-cursed way to use `operator[]` to initialize value.

Reply:
Using of operator[] is a very simple and straightforward part of the C++ language. Unlike templates, SFINAE and move semantics.

Reply:
`wanted_hooks[str]` returns a reference on existing or newly created (by default constructor) value. In this case value is a std::vector of plugin_target_config_entry_t elements.

Reply:
[This](https://youtu.be/lkgszkPnV8g?t=424) is the reason why I don't think `operator[]` is a good thing to use.

Reply:
Louis makes very strange arguments. "Knives should be banned because you can get hurt with a knife." 

I agree that the std::map interface is inconvenient in some use cases, especially when used inappropriately. But this does not mean that it should be banned. We do not prohibit professionals from flying on planes because a non-professional may crash.

Reply:
I agree, but even professionals make mistakes and switching to safer version doesn't cost much, while providing safety and possibly saving hours of debugging in the future.

Reply:
I am always for safe design. I love Haskell and his concepts.

I can implement wanted_hooks_t as a class with good and secure interface, but with a map of vectors inside.

Reply:
Done